### DISCUSSION TITLE
When I make an api request in the Next.js server side area, I cannot get the cookie value

### DISCUSSION THREAD

**Discussion by selcukgiray** [vercel/next.js#68415] [Help]
### Summary

What I want to do: I send a request to the backend in api/auth/me/route.js. I send the token to the backend with **"Authorization: Bearer ${token}"**. As a result, the user information is returned to me successfully.

**api/auth/me/route.js**

```js
import { cookies } from "next/headers";
import { NextResponse } from "next/server";

export async function GET() {
    const cookieStore = cookies();
    const { value: token } = cookieStore.get("token") ?? { value: null };
    const response = await fetch(process.env.API_URL + "/me", {
        method: "GET",
        headers: {
            Accept: "application/json",
            Authorization: `Bearer ${token}`,
        },
    });

    const result = await response.json();

    return NextResponse.json(result);
}
```

"http://localhost:3000/api/auth/me" As you can see, I receive user information.

![image](https://github.com/user-attachments/assets/f443a992-c215-444c-baaf-952cdc217e20)

On my page, I send a request to the api where I get the user information. The response I get back is { message: 'Unauthenticated.' } like this.

{ message: 'Unauthenticated.' } : this is the response I get when the token does not go to the backend or goes wrong.

```js
import React from "react";

async function getData() {
    const res = await fetch("http://localhost:3000/api/auth/me");

    if (!res.ok) {
        throw new Error("Failed to fetch data");
    }

    return res.json();
}

export default async function Page() {
    const data = await getData();
    console.log(data);
    return <div>home</div>;
}
```

http://localhost:3000/api/auth/me, when I request this address on the client side, the data comes without any problems, but I cannot get the data on the server side. 

http://localhost:3000/api/auth/me, when I enter, I can see the user information coming, when I send a request on the server side, I cannot get the data.

**Comment by Juneezee** [comment]
This is because cookies are stored on the browser, not the server. You can see this in your browser developer console

![image](https://github.com/user-attachments/assets/9c22c91e-7d9d-4117-bb59-e4917e4d7b10)


Your `page.tsx` is a server component. It executes `getData()` on the server, which has no cookies.

**Reply by selcukgiray** [reply]
Then how can I get user information on the server side? Will the Redux toolkit etc. help me?

**Reply by Juneezee** [reply]
> Will the Redux toolkit etc. help me?

No.

> Then how can I get user information on the server side?

I would suggest you to read the [Client-server model](https://en.wikipedia.org/wiki/Client%E2%80%93server_model). It is the client (your web browser) that will be sending the requests to the server, not the other way around.

You server cannot perform an action on its own. A client must first send a request to the server, "asking" the server to do something

**Comment by selcukgiray** [comment]
hooks/useAuth.js

```js
import React from "react";

const getUser = async () => {
        const cookies = require("next/headers").cookies;
	const { value: token } = cookies().get("token") ?? { value: null };
	const response = await fetch(process.env.API_URL + "/me", {
		method: "GET",
		headers: {
			Accept: "application/json",
			Authorization: `Bearer ${token}`,
		},
	});

	const result = await response.json();
	return result;
};

// server side

const fromServer = async () => {
	const userData = await getUser();
	return userData;
};

// client

export function useAuth() {
	const [user, setUser] = React.useState({});

	const getUserData = async () => {
		const userData = await getUser();
		setUser(userData);
	};

	React.useEffect(() => {
		getUserData();
	}, []);

	return user;
}

useAuth.fromServer = fromServer;
```

page.js

```js
import { useAuth } from "@/hooks/useAuth";
import React from "react";

export default async function Page() {
	const { name, surname } = await useAuth.fromServer();
	console.log(name, surname);

	return <div>home</div>;
}
```

This way I got the user information and was able to use it on the server side.

The only difference is that I did not use route handler, I sent a request directly to the endpoint.
